Frameworks/Foundation/NSString.mm

```diff
@@ -1093,7 +1097,15 @@ - (NSRange)rangeOfString:(NSString*)searchString options:(NSStringCompareOptions
  @Status Interoperable
 */
 - (BOOL)isAbsolutePath {
-    return ([self hasPrefix:_NSGetSlashStr()]) || ((_isLetter([self characterAtIndex:0])) && ([self characterAtIndex:1] == ':'));
+    if ([self hasPrefix:_NSGetSlashStr()]) {
+        return YES;
+    }
+
+    if (self.length >= 2) {
+        return _isLetter([self characterAtIndex:0]) && [self characterAtIndex:1] == ':';
```

Comment:
This function is technically wrong, because `C:Hello.txt` is a relative path, not an absolute one.

Fixing it is out-of-scope for this change.